modules/calib3d/src/calibration.cpp

```diff
@@ -1529,7 +1628,35 @@ CV_IMPL double cvCalibrateCamera2( const CvMat* objectPoints,
         std::copy(param + 4, param + 4 + 14, k);
 
         if( !proceed )
+        {
+            //do errors estimation
+            if(stdDevs)
+            {
+                Ptr<CvMat> JtJ(cvCreateMat(nparams, nparams, CV_64F));
+                CvMat cvMatM(matM);
+                cvEvaluateJtJ2(JtJ, &matA, &_k, &cvMatM, solver.param, npoints, flags, NINTRINSIC, aspectRatio);
```

Comment:
In most cases solver does step and stops iterations right after it so JtJ value will be outdated here.